### ISSUE TITLE
Gamepad issues (Steelseries Stratus)

### ISSUE THREAD

**Issue by gimblll** [glfw/glfw#1385]
I encountered a few issues on MacOS (Mojave) (up to date glfw from GitHub) with a specific controller (Steelseries Stratus XL) when trying to use it through the gamepad API. The controller has a mapping in the default mappings file, but glfw rejects it because it cannot find all the buttons when initializing the controller. The attached diffs show the needed additions to get it working. (The kHIDPage_Consumer is for the pause button). You can see that SDL uses the same enumerations (+ a few more that might be useful to add) https://www.libsdl.org/tmp/SDL/src/joystick/darwin/SDL_sysjoystick.c

There is also another issue with it. The buttons on this controller are force sensitive, so _GLFWjoyelementNS::maximum is 255 for all the buttons except the pause button. But glfw seems to be passing this number through as is. This then no longer matches with the documented GLFW_PRESS enumeration when reading from GLFWgamepadstate. So the value should probably be clamped to 1, unless you want to support force sensitive buttons?

```
Index: cocoa_joystick.m
===================================================================
--- cocoa_joystick.m	(revision 5570)
+++ cocoa_joystick.m	(revision 5571)
@@ -220,10 +220,18 @@
                 case kHIDUsage_GD_Hatswitch:
                     target = hats;
                     break;
+				case kHIDUsage_GD_DPadUp:
+				case kHIDUsage_GD_DPadDown:
+				case kHIDUsage_GD_DPadLeft:
+				case kHIDUsage_GD_DPadRight:
+					target = buttons;
+					break;
             }
         }
         else if (page == kHIDPage_Button)
             target = buttons;
+		else if (page == kHIDPage_Consumer)
+			target = buttons;
 
         if (target)
         {

```

**Comment by elmindreda** [comment]
Thank you, will review.

**Comment by elmindreda** [comment]
This should be fixed now with 51ca41dd9fdf98bc9d38810f84ad4cb218ecd88a and 463ef7eb71268f57790fdb67f26d328b45d3346e. Thank you!
